src/main/java/seedu/address/logic/commands/TagCommand.java

```diff
@@ -0,0 +1,107 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.commands.EditCommand.EditPersonDescriptor;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Address;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Adds a new tag to the details of an existing person in the address book.
+ */
+public class TagCommand extends Command {
+    public static final String COMMAND_WORD = "tag";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a Tag to the details of the person identified "
+            + "by the index number used in the displayed person list.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_TAG + "TAG" + " [" + PREFIX_TAG + "EXTRA_TAG]...\n"
+            + "Example: " + COMMAND_WORD + " 1 " + PREFIX_TAG + "ExampleTag";
+
+    public static final String MESSAGE_EDIT_PERSON_SUCCESS = "Added Tag to Person: %1$s";
```

Comment:
Should this be MESSAGE_TAG_PERSON_SUCCESS instead?

Reply:
Also not sure how viable is this but should connections display the tags that are being added as well?
Maybe if there is too many tags that are being added at once, display the first 5 and truncate the rest

Reply:
Ok makes sense!